Source/Generator.cs

```diff
@@ -384,173 +415,204 @@ public static ConfigNode GenerateBody(Planet planet, String folder, String refer
-                    Single h = Mathf.Clamp01((Single) ((builddata.vertHeight - pqsVersion.radius) * (1d / pqsVersion.radiusMax)));
-                    diffuse.SetPixel(i, j, builddata.vertColor);
-                    height.SetPixel(i, j, new Color(h, h, h));
-                }
-            }
-            diffuse.UnlockBitmap();
-            height.UnlockBitmap();
+                        for (var j = 0; j < (width / 2); j++)
+                        {
+                            // Create a VertexBuildData
+                            var builddata = new VertexBuildData
+                                            {
+                                                directionFromCenter =
+                                                    Quaternion.CreateFromAngleAxis((360d / width) * i, Vector3.Up) *
+                                                    Quaternion.CreateFromAngleAxis(90d - ((180d / (width / 2.0)) * j),
+                                                                                   Vector3.Right) * Vector3.Forward,
+                                                vertHeight = pqsVersion.radius
+                                            };
+
+                            // Build the maps
+                            pqsVersion.OnVertexBuildHeight(builddata);
+                            pqsVersion.OnVertexBuild(builddata);
+                            builddata.vertColor.a = 1f;
+                            var h = Mathf.Clamp01((float) ((builddata.vertHeight - pqsVersion.radius) *
+                                                           (1d / pqsVersion.radiusMax)));
+                            diffuse.SetPixel(i, j, builddata.vertColor);
+                            height.SetPixel(i, j, new Color(h, h, h));
+                        }
+                    }
+                    diffuse.UnlockBitmap();
+                    height.UnlockBitmap();
 
-            // Save the textures
-            Directory.CreateDirectory(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/");
-            UnsafeBitmap normals = Utility.BumpToNormalMap(height, 9); // TODO: Implement something to make strength dynamic
-            diffuse.Bitmap.Save(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/" + name + "_Texture.png", ImageFormat.Png);
-            height.Bitmap.Save(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/" + name + "_Height.png", ImageFormat.Png); // In case you need it :)
-            normals.Bitmap.Save(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/" + name + "_Normals.png", ImageFormat.Png);
+                    // Save the textures
+                    Directory.CreateDirectory(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/");
+                    using (var normals = Utility.BumpToNormalMap(height, 9)
+                    ) // TODO: Implement something to make strength dynamic
```

Comment:
In the code I marked, there is a new line for a `)`, I don't think that is neccessary. The comment can stay on the new line though.